src/icalendar/prop/__init__.py

```diff
@@ -1808,6 +1790,100 @@ def from_ical(cls, ical):
         except Exception as e:
             raise ValueError(f"Expected , got: {ical}") from e
 
+    @property
+    def ical_value(self) -> str:
+        """The URI."""
+        return self.uri
+
+    @property
+    def uri(self) -> str:
+        """The URI."""
+        return str(self)
+
+    def __repr__(self) -> str:
+        """repr(self)"""
+        return f"{self.__class__.__name__}({self.uri!r})"
+
+    from icalendar.param import FMTTYPE, GAP, LABEL, LANGUAGE, LINKREL, RELTYPE, VALUE
+
+
+class vUid(vText):
+    """A UID of a component.
+
+    This is defined in :rfc:`9253`, Section 7.
+    """
+
+    @classmethod
+    def new(cls):
+        """Create a new UID for convenience.
+
+        .. code-block:: pycon
+
+            >>> from icalendar import vUid
+            >>> vUid.new()
+            vUid('d755cef5-2311-46ed-a0e1-6733c9e15c63')
+
+        """
+        return vUid(uuid.uuid4())
+
+    def __init__(self, uid: str):
+        """Create a vUid."""
+        super().__init__()
+        self.params.setdefault("VALUE", "UID")
+
+    @property
+    def uid(self) -> str:
+        """The uid of this property."""
+        return str(self)
+
+    @property
+    def ical_value(self) -> str:
+        """The uid of this property."""
+        return self.uid
+
+    def __repr__(self) -> str:
+        """repr(self)"""
+        return f"{self.__class__.__name__}({self.uid!r})"
+
+    from icalendar.param import FMTTYPE, LABEL, LINKREL
+
+
+class vXmlReference(vUri):
+    """An XML-REFERENCE.
+
+    The associated value references an associated XML artifact and
+    is a URI with an XPointer anchor value.
+
+    This is defined in :rfc:`9253`, Section 7.
+    """
+
+    def __init__(self, xml_reference: str):
+        """Create a new XML reference."""
+        super().__init__()
+        self.params.setdefault("VALUE", "XML-REFERENCE")
+
+    @property
+    def xml_reference(self) -> str:
+        """The uid of this property."""
+        return self.uri
```

Comment:
```suggestion
    @property
    def xml_reference(self) -> str:
        """The XML reference URI of this property."""
        return self.uri
```

Docstring says "The uid of this property" but should say "The XML reference URI of this property".